### ISSUE TITLE
lint: add range loop suggesting checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#74]
If particular `for` loop can be rewritten in form of `range` loop, suggest it.

This code could be rewritten, so it should trigger a warning:
```go
func sum(xs *[count]int) int {
    res := 0
    for i := 0; i < count; i++ {
        res += xs[i]
    }
    return res
}
```
There are at least 2 suggested forms:
```go
func sumRangeA(xs *[count]int) int {
    res := 0
    for i := range xs {
        res += xs[i]
    }
    return res
}
 
func sumRangeB(xs *[count]int) int {
    res := 0
    for _, x := range xs {
        res += x
    }
    return res
}
```

**Comment by fexolm** [comment]
@Quasilyte, possible duplicate of #72?

**Comment by Quasilyte** [comment]
Not exactly. Referenced issue turns range loops into other range loops.
This is about C-style for loops.

**Comment by Quasilyte** [comment]
We can handle it in one checker or in different checkers, if there is close to 0 shared code.
Hard to tell without implementation.
(Note that `range` and simple for are different AST nodes.)

**Comment by fexolm** [comment]
Ok, clear now.

**Comment by cristaloleg** [comment]
Different checkers, pretty please.
